cli/internal/logstreamer/logstreamer.go

```diff
@@ -125,9 +125,33 @@ func (l *Logstreamer) out(str string) {
 		str = l.colorOkay + l.prefix + l.colorReset + " " + str
 	} else if l.prefix == "stderr" {
 		str = l.colorFail + l.prefix + l.colorReset + " " + str
-	} else {
-		str = l.prefix + str
 	}
 
 	l.Logger.Print(str)
 }
+
+// PrettyStdoutWriter wraps an ioWriter so it can add string
+// prefixes to every message it writes to stdout.
+type PrettyStdoutWriter struct {
+	w      io.Writer
+	Prefix string
+}
+
+// NewPrettyStdoutWriter returns an instance of PrettyStdoutWriter
+func NewPrettyStdoutWriter(prefix string) *PrettyStdoutWriter {
+	return &PrettyStdoutWriter{
+		w:      os.Stdout,
+		Prefix: prefix,
+	}
+}
+
+func (psw *PrettyStdoutWriter) Write(p []byte) (int, error) {
```

Comment:
This feels like an improper implementation of `io.Write`. For correct behavior it requires `p` to end in a newline and if it ends up getting wrapped by `bufio.Writer` prefixes will almost certainly not be written where they're expected. 

I think keeping around some internal state for tracking when a newline has been encountered and only adding the prefix then might be necessary.

Reply:
I could be wrong, but I think `Logstreamer` already handles this. It only calls `.out()` (which is how the multiwriter and this io.writer wrapper gets called) when on new lines

https://github.com/vercel/turborepo/blob/ce2e47b6b508db7038d127416a052b0da02e11ba/cli/internal/logstreamer/logstreamer.go#L85-L93

the general setup is:

1. `Logstreamer` has a `Logger`
2. `Logger` is a stdlib `log.New()` instance with a `.writer` that is a `io.multiwriter`
3. `io.multiwriter` has this pretty writer.

Reply:
You're right and I think that's probably where this logic should live. My concern is the unintended consequences if the `Logstreamer` was changed or someone tried to use `PrettyStdoutWriter` raw. Probably out of scope since it would require larger refactor.

Reply:
It's a good call out. The number of logging interfaces in general are pretty confusing to me, so the larger reactor could include cleaning that up too.